src/v/redpanda/admin/api-doc/debug.json

```diff
@@ -325,9 +325,53 @@
                     ]
                 }
             ]
+        },
+        {
+            "path": "/v1/debug/cpu_profile",
+            "operations": [
+                {
+                    "method": "GET",
+                    "summary": "Gets the samples from the CPU profiler",
+                    "nickname": "cpu_profile",
+                    "produces": [
+                        "application/json"
+                    ],
+                    "type": "array",
+                    "items": {
+                        "type": "cpu_profile_sample"
+                    },
+                    "parameters": [
+                        {
+                            "name": "shard",
+                            "in": "query",
+                            "required": false,
+                            "allowMultiple": false,
+                            "type": "long"
+                        }
+                    ]
+                }
+            ]
         }
     ],
     "models": {
+        "cpu_profile_sample": {
+            "id": "cpu_profile_sample",
+            "description": "cpu profile sample",
+            "properties": {
+                "shard_id": {
```

Comment:
Any specific reason you didn't group by shard id in a first instance?

Reply:
Group by shard as in why I'm returning per-shard information at all? Or why the return format isn't more like;
```json
[
    {
        "shard_id": 0,
        "backtraces": [...]
     },
     {
        "shard_id": 1,
        "backtraces: [...]
     },
     ...
]
```

If it's the first one then because of RP's share nothing architecture different threads are close to behaving like different processes and can have completely independent profiles. So it's important to provide a view of each independently.

If it's  the second then it's because returning them in this flat format was easier and saved a few allocations. The client can always group things together however it works best for a particular use case.

Reply:
> If it's the second then it's because returning them in this flat format was easier and saved a few allocations. The client can always group things together however it works best for a particular use case.

Yeah, that's what I meant. Fair, was just wondering as the service stores them per shard already anyway.

Reply:
Switched to
```
[
    {
        "shard_id": 0,
        "backtraces": [...]
     },
     {
        "shard_id": 1,
        "backtraces: [...]
     },
     ...
]
```
Similar to how you did it in the memory profiler. It ended up making more sense.

Reply:
Seems like this is resolved.